components/board/interrupt_stream.go

```diff
@@ -0,0 +1,111 @@
+package board
+
+import (
+	"context"
+	"sync"
+
+	pb "go.viam.com/api/component/board/v1"
+	"go.viam.com/utils"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type interruptStream struct {
+	*client
+	streamCancel  context.CancelFunc
+	streamRunning bool
+	streamReady   chan bool
+	streamMu      sync.Mutex
+
+	activeBackgroundWorkers sync.WaitGroup
+	extra                   *structpb.Struct
+}
+
+func (s *interruptStream) startStream(ctx context.Context, interrupts []string, ch chan Tick) error {
+	s.streamMu.Lock()
+	defer s.streamMu.Unlock()
+
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	s.streamRunning = true
+	s.streamReady = make(chan bool)
+	s.activeBackgroundWorkers.Add(1)
+	ctx, cancel := context.WithCancel(ctx)
+	s.streamCancel = cancel
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+	}
+
+	req := &pb.StreamTicksRequest{
+		Name:     s.client.info.name,
+		PinNames: interrupts,
+		Extra:    s.extra,
+	}
+
+	// This call won't return any errors it had until the client tries to receive.
+	//nolint:errcheck
+	stream, _ := s.client.client.StreamTicks(ctx, req)
+	_, err := stream.Recv()
+	if err != nil {
+		s.client.logger.CError(ctx, err)
+		return err
+	}
+
+	// Create a background go routine to receive from the server stream.
```

Comment:
```suggestion
	// Create a background go routine to receive from the server stream. We rely on calling the Done function here rather than in close stream since managed go calls that function when the routine exits.
```